### PR TITLE
Feature/dropdown menu

### PR DESCRIPTION
**Pull request by HarryLee87**
**Jira Ticket**<br>
Link: https://derek-team-lva74agprdbm.atlassian.net/browse/SP-69

**Description**<br>
_Add a sensible description to this pull request:_
With the mock category array, it displays the category when it is clicked.
Additionally, I added fetching category name data from backend to display on the category dropdown menu.
Lastly, I fixed the adjustable search bar length due to the length of category's button by assigning a specific percentage of the basis.

- Added: useCategories.ts
- Changed: Navigation.tsx
- refactored: useCategories. (fetch category name from backend)

**Screenshots**<br>
image

### CODE REVIEW

File: components/layouts/Navigation.tsx [hunk 1]
```diff
@@ -4,8 +4,21 @@ import Image from "next/image";
...
 import { LiaShoppingCartSolid } from "react-icons/lia";
+import useCategories from "@/hooks/useCategories";
+import { useState } from "react";
 
 export default function NavigationBar() {
+  const { categories, loading, error } = useCategories();
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const handleCategorySelect = (category: string) => {
+    console.log(`Selected category: ${category}`);
```
Comment (derekhskim) [inline components/layouts/Navigation.tsx]: Please remove console,log code when in stage of merging to main branch

**Comment by derekhskim** [comment]
User should also be able to select category and see the changes that was reflected

File: yarn.lock
Comment (jay-kj-woo) [inline yarn.lock:1]: Wonder why yarn.lock has been modified.... 🤔

Comment (HarryLee87) [reply yarn.lock:1]: As i have checked the yarn.lock file, it looks updated. Would it be some issue, if we merge with updated yarn.lock?

Comment (HarryLee87) [reply yarn.lock:1]: > Wonder why yarn.lock has been modified.... 🤔
As i have checked the yarn.lock file, it looks updated. Would it be some issue, if we merge with updated yarn.lock?

Comment (jay-kj-woo) [reply yarn.lock:1]: I don't think we are using the `immutable` option in our CI/CD so it shouldn't be a problem. Usually, we would like to freeze our lock file or make it immutable to ensure the same environment among the developers.

**Comment by HarryLee87** [comment]
> User should also be able to select category and see the changes that was reflected

Changed the selected category to be displayed.

File: yarn.lock
Comment (HarryLee87) [reply yarn.lock:1]: > immutable

It is just minor updates for now. I will look for how to change the immutable option in our CI/CD soon

File: components/layouts/Navigation.tsx (hunk 1 above)
Comment (HarryLee87) [reply components/layouts/Navigation.tsx]: > Please remove console,log code when in stage of merging to main branch

I removed the console.log code.

**Review by Euna-kim-1 — APPROVED** [review]

**Pull request merged**